tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py

```diff
@@ -0,0 +1,191 @@
+import time
+from ducktape.mark import matrix
+from ducktape.tests.test import Test
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.version import LATEST_0_10_0, LATEST_0_10_1, LATEST_0_10_2, LATEST_0_11_0, LATEST_1_0, LATEST_1_1, \
+    LATEST_2_0, LATEST_2_1, LATEST_2_2, LATEST_2_3, DEV_BRANCH, DEV_VERSION, KafkaVersion
+from kafkatest.services.streams import CooperativeRebalanceUpgradeService
+from kafkatest.tests.streams.utils import verify_stopped, stop_processors, verify_running
+
+
+class StreamsCooperativeRebalanceUpgradeTest(Test):
+    """
+    Test of a rolling upgrade from eager rebalance to
+    cooperative rebalance
+    """
+
+    source_topic = "source"
+    sink_topic = "sink"
+    task_delimiter = "#"
```

Comment:
Is it necessary to make `TASK_DELIMITER` configurable? I feel it is unnecessarily complicating the setup..

Reply:
Ack, although the test uses this delimiter (or will in the near future) so I'll leave the definition here but remove from the configs passed.

Reply:
Yeah my point here is that we can just hard-coded it than passing it through.